### ISSUE TITLE
Dataflow exploration for state of LA

### ISSUE THREAD

**Issue by scleary1cs** [CDCgov/trusted-intermediary#580]
# Story
As a CDC TI team, so that we can explore all data flow options we need to see if we can utilize the existing data flow infrastructure between RS and the State Health Department.

data flow: the flow of HL7 messages to a PHL; can we re-use existing architecture/infrastructure to get the messages to where they need to go

## Pre-conditions
- [ ] *Assumptions of prior or future work that's out of scope for this story*

## Acceptance Criteria
- [x] Determine whether data will be routed through the DOH or directly to the PHL
- [x] A LA-specific dataflow that is accepted and approved by the site partners

### Research
- [ ] *Research work needed to complete the story*
- [ ] Foundational: *High-level research that will support this and future efforts*

###  Engineering
- [ ] *Engineering work needed to complete the story*
- [ ] Foundational: *Technical runway work to support this and future efforts*

## Definition of Done
- [ ] Documentation tasks completed
  - [ ] Documentation and diagrams created or updated
    - [ ] Implementation guide (`/ig` folder)
    - [ ] ADRs (`/adr` folder)
    - [ ] Main `README.md`
    - [ ] Other READMEs in the repo
    - [ ] If applicable, update the `ReportStream Setup` section in `README.md`
  - [ ] Threat model updated
  - [ ] API documentation updated
  - [ ] Source code documentation created when the code is not self-documenting
- [ ] Code quality tasks completed
  - [ ] Logging added where useful
  - [ ] Code refactored for clarity and no design/technical debt
  - [ ] Adhere to separation of concerns; code is not tightly coupled, especially to 3rd party dependencies
  - [ ] Code is reviewed or developed by pair; 1 approval is needed but consider requiring an outside-the-pair reviewer
  - [ ] Code quality checks passed
- [ ] Security & Privacy tasks completed
  - [ ] Security & privacy gates passed
- [ ] Testing tasks completed
  - [ ] Load tests passed
  - [ ] Unit test coverage of our code >= 90%
- [ ] Build & Deploy tasks completed
  - [ ] Build process updated
  - [ ] API(s) are versioned
  - [ ] Feature toggles created and/or deleted.  Document the feature toggle
  - [ ] Source code is merged to the main branch

## Research Questions
- *Optional: Any initial questions for research*

## Decisions
- *Optional: Any decisions we've made while working on this story*

## Notes
- Newborn Screening ETOR w/ CDC (meeting with LA DOH and re-using their RS connection)
  - Went over current state flow and our options.
  - I proposed a strawman proposal of putting our files into a separate folder on the existing SFTP site.  MoveIt then is used to transfer our files to Natus, or if required due to some policy, MoveIt moves files to OPH Labs SFTP site and then to Natus.
  - In general, yes, that should work.
  - A lot of requirements for MoveIt
    - File level encryption using RSA public/private key pair, but could also do AES256.
    - Can get an exception to this.
    - Key pair needs to expire in 5 years.
    - Key pair per environment.
    - Key pair e-mail needs to be a group e-mail, not an individual.
  - Brad Coney, of the LA cybersecurity team, said this sounds complicated and has a lot of steps.  Let's explore using less hops to get the information to Natus.  They are already pursuing this avenue given yesterday's meeting.
    - Many others agreed that this sounds extra complicated.
  - (Unsure if that means they are pursuing a policy exception for us.)
  - Decision: Brad Coney still looking into things, and if an LA SFTP site is required, it will be a new one instead of reusing the existing LA DOH RS connection.
  - Hot take: MoveIt and its requirements sounds like something that is out of our control.  It is something that LA DOH needs to implement, not us.  RS, today, isn't doing any of those requirements; they are just dropping files off in a LA DOH SFTP site, so RS shouldn't do it for ETOR either.

**Comment by sfradkin** [comment]
Waiting on hearing back from LA OTS.  Adrian out until next week.

**Comment by Adrian-Brewster** [comment]
After meeting with LA DOH and PHL, we determined routing ETOR through the DOH was not feasible at this time.  Will continue to explore this model for future partners.
